src/nested_pandas/nestedframe/io.py

```diff
@@ -136,8 +136,16 @@ def read_parquet(
             field_names = [table.column_names[i] for i in indices]
 
             # Use iterchunks to process chunks of each column
-            chunked_arrays = [pa.concat_arrays(list(table.column(i).iterchunks())) for i in indices]
-
+            chunked_arrays = []
+            for i in indices:
+                column = table.column(i)
+                if len(column.chunks) == 1:
+                    # If there is only one chunk, use it directly
+                    # avoid copy in concat_arrays
+                    chunked_arrays.append(column.chunk(0))
+                else:
+                    # Otherwise, concatenate all chunks
+                    chunked_arrays.append(pa.concat_arrays(list(column.iterchunks())))
```

Comment:
I'd prefer to never concatenate chunks, let's maybe adopt the same approach as used here?
https://github.com/lincc-frameworks/nested-pandas/blob/2f8cf4cf232632b9c8bd8ed69c0091d09b016611/src/nested_pandas/series/ext_array.py#L859

Reply:
Gotcha, I started looking into that approach, but stumbled upon a one-liner solution that seems to do everything we want: `table.select(indices).to_struct_array()`